File: ruby/recursive_count.rb

# count = 0

# while count < 10
#   puts count
#   count += 1
# end
def recursive_count(num = 0)
  return if num > 9
  puts num
  recursive_count(num += 1)
end

if __FILE__ == $PROGRAM_NAME
  recursive_count
end

# OPTIONAL
# Please add your pseudocode to this file
# And a written explanation of your solution

File: javascript/recursive_count.js
// let count = 0;

// while (count < 10) {
//   console.log(count);
//   ++count;
// }

/*
  return if count hits 10 (is greater than 9)
  log the count
  call fn, pass it incremented count value
*/

function recursiveCount(num = 0) {
  if (num === 10) return 
  console.log(num)
  recursiveCount(num += 1)
}

if (require.main === module) {
  recursiveCount();
}

module.exports = recursiveCount;

// OPTIONAL
// Please add your pseudocode to this file
// And a written explanation of your solution
